### PR TITLE
Fix support for bignum array

### PR DESCRIPTION
**Pull request by sydhds**

### CODE REVIEW

**Review by peterjah — CHANGES_REQUESTED** [review]
a lot of code duplication.
I think it would have been possible to keep a generic templated function.
the byte buffer size of a class car be retrieved with `offsetof<T>()` 

You keeped `nextFixedSizeArray` but we now its not working with some types, maybe adding a check on types would be great to avoid futur bugs.

it would be also nice to complete/add unit tests that cover at least the bug you are fixing,

**Comment by sydhds** [comment]
> a lot of code duplication. I think it would have been possible to keep a generic templated function. the byte buffer size of a class can be retrieved with `offsetof<T>()`
> 
> You keeped `nextFixedSizeArray` but we know its not working with some types, maybe adding a check on types would be great to avoid future bugs.
> 
> it would be also nice to complete/add unit tests that cover at least the bug you are fixing,

I've tried the generic function but the compiler rejects it. I've updated the code to use offsetof() and updated the unit tests as well.

**Review by peterjah — APPROVED** [review]

**Review by sydhds — COMMENTED** [review]
I would remove support for i256 as it is almost unimplemented in as-bignum...

File: packages/as-types/assembly/__tests__/argument.spec.ts [hunk 1]
```diff
@@ -496,23 +504,4 @@ describe('Args tests', () => {
     expect(deser[1].age).toBe(45);
     expect(deser[1].name).toBe('Superman');
   });
-
-  it('With array of bytes', () => {
```
[the remark this answers is not in the document]
Comment (peterjah) [reply packages/as-types/assembly/__tests__/argument.spec.ts:500]: Yes i removed the support of serialize array of fixed size StaticArray... i don't see any usecase to this.
And its may be source of bug if the user provide elements of different sizes...

File: packages/as-types/assembly/serialization/bignum.ts [hunk 2]
```diff
@@ -12,6 +13,17 @@ export function bytesToU128(arr: StaticArray<u8>): u128 {
...
 
+/**
+ * Converts a StaticArray<u8> into a i128.
+ *
+ * @param arr - the array to convert
+ *
+ * @returns the converted i128
+ */
+export function bytesToI128(arr: StaticArray<u8>): i128 {
+  return changetype<i128>(u128.from(wrapStaticArray(arr)));
```
[the remark this answers is not in the document]
Comment (peterjah) [reply packages/as-types/assembly/serialization/bignum.ts:24]: i128 use same serialization of u128. its just the way we interprets the bytes that change

**Pull request merged**